### DISCUSSION TITLE
OffsetDateTime::parse -> format question

### DISCUSSION THREAD

**Discussion by kgrodzicki** [time-rs/time#470] [Question] [answered]
Hi :)

I have a question regarding parsing and formatting the OffsetDateTime.

The test below shows the case where parse and then format is used using Rfc3339 format. The result string is 2013-10-07T04:23:19.12Z instead of 2013-10-07T04:23:19.120Z.

```
#[test]
fn test_parse() {
  let date_time = "2013-10-07T04:23:19.120Z";
  let created_at = OffsetDateTime::parse(date_time, &Rfc3339).unwrap();
  assert_eq!(date_time, created_at.format(&Rfc3339).unwrap());
}
```

Using:
time = { version="0.3.9", features = ["serde-well-known", "parsing", "formatting"] }

/Krzysztof

**Comment by jhpratt** [comment, marked as answer]
The `OffsetDateTime` struct doesn't know how many digits were parsed; formatting and parsing are two separate operations. When formatting, the fewest number of digits necessary is used. In this case that is two.

**Reply by kgrodzicki** [reply]
@jhpratt Thank you for your fast answer! Is there a way to preserve this information using time lib?

**Reply by jhpratt** [reply]
Not built-in, no. The reason being that formatting and parsing are two separate operations. You can create a custom format description if you'd like, but that won't be compliant with the RFC3339 specification (which is bizarrely complicated).
